### PR TITLE
Allow editing of some AI settings in the admin pages

### PR DESCRIPTION
**Pull request by marcospri**
Mistakes while we input these couldn't be solved, allow editing them.

This is more relevant now with the LTI1.3 upgrade flow were these might need updating anyway.

### CODE REVIEW

File: lms/views/admin/application_instance.py [hunk 1]
```diff
@@ -59,7 +59,7 @@ class NewApplicationInstanceSchema(PyramidRequestSchema):
     email = fields.Email(required=True)
 
 
-class UpdateApplicationInstanceSchema(ApplicationInstanceSchema):
+class UpgradeApplicationInstanceSchema(ApplicationInstanceSchema):
```
Comment (marcospri) [inline lms/views/admin/application_instance.py:62]: This schema is not for updating the AI but for upgrading it to LTI1.3.

File: lms/views/admin/application_instance.py [hunk 2]
```diff
@@ -265,6 +265,10 @@ def show_instance(self):
     def update_instance(self):
         ai = self._get_ai_or_404(**self.request.matchdict)
 
+        for ai_field in ["lms_url", "custom_canvas_api_domain", "deployment_id"]:
+            if value := self.request.params.get(ai_field):
+                setattr(ai, ai_field, value.strip())
```
Comment (jon-betts) [inline lms/views/admin/application_instance.py:270]:
```suggestion
            if value := self.request.params.get(ai_field, "").strip():
                setattr(ai, ai_field, value)
```

Might be a better way than this, but a string containing only spaces will be truthy until stripped.

File: lms/templates/admin/instance.html.jinja2 [hunk 3]
```diff
@@ -42,8 +42,8 @@
 
     <fieldset class="box">
         {{ macros.disabled_text_field("Consumer key", instance.consumer_key) }}
-        {{ macros.disabled_text_field("Deployment ID", instance.deployment_id) }}
-        {{ macros.disabled_text_field("LMS URL", instance.lms_url) }}
+        {{ macros.form_text_field(request, "Deployment ID", "deployment_id", field_value=instance.deployment_id) }}
+        {{ macros.form_text_field(request, "LMS URL", "lms_url", field_value=instance.lms_url) }}
```
Comment (jon-betts) [inline lms/templates/admin/instance.html.jinja2:46]: I thought this was one of the fields we automatically updated on application instances, but it's not. I wonder if we could / should, as this field is probably very wrong in most cases.

File: tests/unit/lms/views/admin/application_instance_test.py [hunk 4]
```diff
@@ -269,6 +269,33 @@ def test_update_instance(
...
         )
 
+    @pytest.mark.parametrize(
+        "setting,value,expected",
+        (
+            ("lms_url", "http://some-url.com", "http://some-url.com"),
+            ("custom_canvas_api_domain", "some.domain.com", "some.domain.com"),
+            ("deployment_id", "DEPLOYMENT_ID", "DEPLOYMENT_ID"),
+        ),
+    )
```
Comment (jon-betts) [inline tests/unit/lms/views/admin/application_instance_test.py:279]: Could do with a few examples to prove what happens when the values have space or are blank etc.

File: lms/views/admin/application_instance.py (hunk 2 above)
Comment (jon-betts) [reply lms/views/admin/application_instance.py:270]: Is there any uniqueness check or anything else for `deployment_id`?

**Review by jon-betts — APPROVED** [review]
The editing work and seems like an obvious improvement. I wonder if we should enable this for other fields too?

One thing that occurred to me is, should there be any kind of check around the deployment id?

File: lms/templates/admin/instance.html.jinja2 (hunk 3 above)
Comment (marcospri) [reply lms/templates/admin/instance.html.jinja2:46]: I removed the option to edit the canvas custom domain, this is overridden in every request from the request parameters.

**Pull request merged**